internal/tests_config.go

```diff
@@ -57,35 +59,90 @@ type Stats struct {
 //
 // May contain prefixes; the longest prefix wins.
 type TestsConfig struct {
-	Default status   `yaml:"default"`
-	Stats   *Stats   `yaml:"stats"`
-	Pass    []string `yaml:"pass"`
-	Skip    []string `yaml:"skip"`
-	Fail    []string `yaml:"fail"`
+	Default status `yaml:"default"`
+	Stats   *Stats `yaml:"stats"`
+	Pass    *Tests `yaml:"pass"`
+	Skip    *Tests `yaml:"skip"`
+	Fail    *Tests `yaml:"fail"`
 }
 
-func (tc *TestsConfig) toMap() (map[string]status, error) {
-	res := make(map[string]status, len(tc.Pass)+len(tc.Skip)+len(tc.Fail))
+type Tests struct {
+	TestNames []string
+	ResRegexp []string
+}
 
-	for _, t := range tc.Pass {
-		if _, ok := res[t]; ok {
-			return nil, fmt.Errorf("duplicate test or prefix: %q", t)
-		}
-		res[t] = Pass
-	}
+type importedTestsConfig struct {
+	Default status `yaml:"default"`
+	Stats   *Stats `yaml:"stats"`
+	Pass    []any  `yaml:"pass"`
+	Skip    []any  `yaml:"skip"`
+	Fail    []any  `yaml:"fail"`
+}
 
-	for _, t := range tc.Skip {
-		if _, ok := res[t]; ok {
-			return nil, fmt.Errorf("duplicate test or prefix: %q", t)
+func (itc *importedTestsConfig) Convert() (*TestsConfig, error) {
+	tc := TestsConfig{itc.Default, itc.Stats, &Tests{}, &Tests{}, &Tests{}}
+	for _, tcat := range []struct {
+		inTests  []any
+		outTests *Tests
+	}{
+		{itc.Pass, tc.Pass},
+		{itc.Skip, tc.Skip},
+		{itc.Fail, tc.Fail},
+	} {
+		for _, t := range tcat.inTests {
+			switch t.(type) {
+			case map[string]any:
+				if _, ok := t.(map[string]any); ok {
+					//TODO
+					continue
+				}
+				panic("map[string]any assertion error")
+			case string:
+				if testname, ok := t.(string); ok {
+					tc.Pass.TestNames = append(tc.Pass.TestNames, testname)
+					continue
+				}
+				panic("string assertion error")
+			default:
+				return nil, fmt.Errorf("invalid type of \"%q\": %q", t, reflect.TypeOf(t))
+			}
```

Comment:
@AlekSi Are you sure about the type switch in this place? Is there any advantage on doing a type assertion twice?